pkg/tools/kubectl_filter.go

```diff
@@ -133,33 +149,39 @@ func analyzeCall(call *syntax.CallExpr) string {
 
 	klog.V(2).Infof("analyzeCall: found kubectl: %q", firstArg)
 
-	// Get the verb (first non-flag argument after kubectl)
-	verbPos := 1 // Start after kubectl at position 0
-	for verbPos < len(args) && strings.HasPrefix(args[verbPos], "-") {
-		verbPos++
+	// Find the verb: first non-flag argument after "kubectl"
+	verb := ""
+	subVerb := ""
+	for i := 1; i < len(args); i++ {
+		if !strings.HasPrefix(args[i], "-") {
+			if verb == "" {
+				verb = args[i]
+			} else {
+				subVerb = args[i]
+				break
+			}
+		}
 	}
-
-	if verbPos >= len(args) {
+	if verb == "" {
 		klog.Warningf("analyzeCall: no verb found after kubectl in args: %v", args)
 		return "unknown"
 	}
 
-	verb := args[verbPos]
 	hasDryRun := hasDryRunFlag(strings.Join(args, " "))
```

Comment:
couldn't resist :)

we also can check for dry-run flags in the same loop all the way and can kill this helper function :)

Reply:
Thanks @droot 

I merged the `verbs` and  `hasDryRun`  into a single loop and was planning to delete this function.

But I noticed this function has a test case with some good cases in it.
So I refactored the loop out and used this test to verify it. What do you think?